src/shared/api/http.ts

```diff
@@ -11,58 +32,69 @@ async function get<T>(path: string, params?: PaginationParams): Promise<T> {
   return res.json();
 }
 
-export function fetchAccounts(params?: PaginationParams): Promise<Paginated<Account>> {
-  return get('/accounts', params);
-}
+export async function fetchAccounts(params?: PaginationParams): Promise<Paginated<Account>> {
+  const res = await get<Paginated<Account>>('/accounts', params);
+  accountSchema.array().parse(res.items);
 
-export function fetchAccount(id: string): Promise<Account> {
-  return get(`/accounts/${id}`);
+  return res;
```

Comment:
1. No need to assign a type to request response (`get<...>`) if you pass it then to Zod's parsing. It accepts `unknown` and produced a typed result.
2. You parse data with schema, but still return non-validated and, more importantly, untransformed response.
3. Define a reusable schema for `Paginated<T>`

```suggestion
  const res = await get('/accounts', params);
  return paginatedSchema(accountSchema).parse(res)
```

```ts
const paginatedSchema = <T extends z.ZodType>(item: T) =>
    z.object({
        pagination: paginationSchema,
        items: item.array()
    })
```